### PR TITLE
Move rake routes task to rails command

### PR DESCRIPTION
**Pull request by benoittgt**
### Summary

After a [discussion with @matthewd](https://github.com/rails/rails/issues/32117). It was mentioned that rake tasks need
to be moved to rails command.

First PR for this subject from me. 😀

### Other Information

- Is it an issue to don't deal anymore with rake params? https://github.com/benoittgt/rails/commit/1ab3ddcfc7efc51ccb206e73c91a67b69a82a6b9#diff-e1c35224a4df7f73d90a9a90e45aab2bL255 and do we need to add a changelog for this?
- Do we want to add more constraints to args?

### CODE REVIEW

File: railties/lib/rails/commands/routes/routes_command.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      def perform(*)
+        require_application_and_environment!
+        require "action_dispatch/routing/inspector"
+
+        all_routes = Rails.application.routes.routes
+        inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
+
+        routes_filter = nil
+
+        if (controller_search = ARGV.index { |arg| arg == "-c" })
```
Comment (rafaelfranca) [inline railties/lib/rails/commands/routes/routes_command.rb]: I think command allow you to define options to not need to build your own option parse for each command. Can you change it to use the options like https://github.com/rails/rails/blob/f7157e48ea233eab56844dba4203f5149d910eaa/railties/lib/rails/commands/server/server_command.rb#L114?

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: railties/lib/rails/commands/routes/routes_command.rb (hunk 1 above)
Comment (benoittgt) [reply railties/lib/rails/commands/routes/routes_command.rb]: Thanks for the review and the tips. I didn't know you can do this. I made the change.

File: railties/lib/rails/commands/routes/routes_command.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      end
+
+      def perform(*)
+        require_application_and_environment!
+        require "action_dispatch/routing/inspector"
+
+        all_routes = Rails.application.routes.routes
+        inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
+
+        puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, routes_filter)
```
Comment (y-yagi) [inline railties/lib/rails/commands/routes/routes_command.rb]: Let's use `say` as with `help`.

File: railties/test/commands/routes_test.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+require "isolation/abstract_unit"
+require "env_helpers"
+require "rails/command"
+require "rails/commands/routes/routes_command"
+
+class Rails::Command::RoutesTest < ActiveSupport::TestCase
+  setup :build_app
+  teardown :teardown_app
+
+  test "test_singular_resource_output_in_rake_routes" do
```
Comment (y-yagi) [inline railties/test/commands/routes_test.rb]: If you use `test` method, you do not need to separate them with `_`. Therefore, you can change this like `singular resource output `.

**Review by rafaelfranca — APPROVED** [review]
Can you squash your commits?

**Comment by benoittgt** [comment]
Thanks a lot for the review. I will try to move other rake tasks.

**Comment by eileencodes** [comment]
@benoittgt when you do move other tasks can you leave the test tasks alone? I'm working on a project with them now and I don't want to have to deal with a ton of conflicts and rewriting 😄

**Comment by benoittgt** [comment]
@eileencodes oups. So to be sure you want me to doesn't remove tests in [`railties/test/application/rake_test.rb`](https://github.com/rails/rails/pull/32121/commits/a2748eda586b11db59e2d0f59ab37910156d6c32#diff-e1c35224a4df7f73d90a9a90e45aab2b)?

Maybe I can revert the rake tests removed for this PR?

**Comment by eileencodes** [comment]
@benoittgt doh I'm sorry it was early and I had not had enough coffee.

I meant the database rake tasks in Active Record. You're all good on Railties. 😄

**Pull request merged**